src/service.rs

```diff
@@ -290,6 +292,14 @@ pub fn new_full(
 									}
 								},
 								Ok(None) => (),
+								Err(pow::Error::Compute(pow::compute::Error::CacheNotAvailable)) => {
+									thread::sleep(Duration::new(1, 0));
+								},
+								Err(pow::Error::Compute(err)) => {
+									warn!("Mining failed: {}", err.description());
+
+									process::exit(libc::EXIT_FAILURE)
```

Comment:
As I see when someone wants to stop kulupu with SIGINT, CTRL+C, just happens that `run_node_until_exit` 'wakes up' and let the main thread reach the end of the main and so exiting. As i see there isn't a clean procedure, so i guess it's fine to exit from here.
I also thought a SIGINT could be sent from here to let `run_node_until_exit` wake up (but i should share somehow the result otherwise it would exist as success)